mqttclient/mqttclient.go

```diff
@@ -0,0 +1,50 @@
+package mqttclient
+
+import (
+	"tinygo.org/x/drivers/net/mqtt"
+)
+
+type Client struct {
+	server     string
+	id         string
+	mqttClient mqtt.Client
+}
+
+func New(server, id string) Client {
```

Comment:
Why not return a Client pointer:

```
func New(server, id string) *Client {
    ....
    return &Client{ ...
```

This way you reference the client instead of pass by value.

Reply:
so this struct is not going to be mutated by the caller, and it's a small struct.

Reply:
However, Client exposes a method that have a side effect, perhaps it's a good idea to have it as a reference: 

```
Taking a pointer signals that this method will modify the object, or have some kind of side effect on it. In this case,
it writes to the attached writer.

-- The Power of Go - Tools by John Arundel
```

Reply:
In fact I was just pointing to your `New()` and `Connect()` functions as those are the ones used externally to instantiate your object, the other methods are most likely ok to not use pointers as you are not setting any value to the struct fields.